code/game/objects/items/weapons/flamethrower.dm

```diff
@@ -15,239 +126,267 @@
+	consumption_rate = 20
+	lit = FALSE
+	tool_behaviour = null
+	toolspeed = 0.5
+	usesound = 'sound/items/welder.ogg'
+	var/consumption_rate_welder = 5
+	var/force_welder = 30
+	var/armour_penetration_welder = 50
+	var/status = 1
+	var/create_with_tank = FALSE
+
+/obj/item/flamethrower/advanced/Destroy()
+	QDEL_NULL(ptank)
+	return ..()
+
+/obj/item/flamethrower/advanced/update_icon()
+	cut_overlays()
+	if(ptank)
+		add_overlay("+ptank")
+		if(ptank.air_contents.toxins > 20) //preempting that runetime.
+			add_overlay("+fuel5")
+		else if(ptank.air_contents.toxins > 15)
+			add_overlay("+fuel4")
+		else if(ptank.air_contents.toxins > 10)
+			add_overlay("+fuel3")
+		else if(ptank.air_contents.toxins > 5)
+			add_overlay("+fuel2")
+		else if(ptank.air_contents.toxins < 5)
+			add_overlay("+fuel1")
+	else 
+		add_overlay("+fuel0")
+	if(lit)
+		item_state = "burner[status]"
+		add_overlay("+lit[status]")
+	else
+		item_state = "burner0"
+		icon_state = "burner[status]"
+	if(ismob(loc))
+		var/mob/M = loc
+		M.update_inv_l_hand()
+		M.update_inv_r_hand()
+
+/obj/item/flamethrower/advanced/attack(mob/living/M, mob/living/user)
+	if(user.get_inactive_hand())
+		to_chat(user, "<span class='userdanger'>You need both hands free to attack with \the [src]!</span>")
+		return
+	if(lit && status == 2) //we want it to have different attack values depending on what status and if it's on or not
+		damtype = BURN
+		force = force_welder
+		armour_penetration = armour_penetration_welder
+		hitsound = 'sound/items/welder.ogg'
+		consume_fuel(consumption_rate)
+	else
+		damtype = BRUTE
+		force = initial(force)
+		armour_penetration = initial(armour_penetration)
+		hitsound = "swing_hit"
+	..()
+
+/obj/item/flamethrower/advanced/afterattack(atom/target, mob/user, flag)
+	. = ..()
+	if(flag)
+		return // too close
+	if(user.get_inactive_hand())
+		to_chat(user, "<span class='userdanger'>You need both hands free to use \the [src]!</span>")
+		return
+	if(lit)
+		if(status == 2)
+			return
+		else if(user && user.get_active_hand() == src) // Make sure our user is still holding us
+			playsound(get_turf(src),'sound/magic/fireball.ogg', 200, TRUE)
+			fireline(target, user)
+			consume_fuel(consumption_rate)
+
+obj/item/flamethrower/advanced/tool_use_check(mob/living/user)
+	if(!lit)
+		to_chat(user, "<span class='notice'>[src] has to be on to complete this task!</span>")
+		return FALSE
+	return TRUE
+
+obj/item/flamethrower/advanced/use_tool(atom/target, mob/living/user, delay, amount, volume, datum/callback/extra_checks)
+	. = ..()
+	consume_fuel(consumption_rate)
+
+/obj/item/flamethrower/advanced/AltClick(mob/user)
+	if(lit)
+		to_chat(user, "<span class='notice'> turn off [src] first!</span>")
+	else if(ptank && isliving(user) && user.Adjacent(src))
+		user.put_in_hands(ptank)
+		ptank = null
+		to_chat(user, "<span class='notice'>You remove the plasma tank from [src]!</span>")
+		update_icon()
+
+obj/item/flamethrower/advanced/CtrlClick(mob/user)
```

Comment:
This does mean you can't drag them anymore. If you include a check to see if the loc of the src is the user then you can still allow the item to be dragged